### PR TITLE
new PR for testing missing elements

### PR DESCRIPTION
**Pull request by RedOrNot**
original PR : https://github.com/ansible/ansible-ui/pull/2549

### CODE REVIEW

File: cypress/support/hostsfunctions.ts [hunk 1]
```diff
@@ -322,3 +335,30 @@ export function checkFactsInHost(inventory: Inventory, hostInInventory?: boolean
   cy.containsBy('a', 'Facts').click();
   cy.get('code').should('contain', 'ansible_dns');
 }
+
+export function testMissingButton(
+  host_type: string,
+  inventory: Inventory,
+  //view: string,
```
Comment (fullsushidev) [inline cypress/support/hostsfunctions.ts:342]: Is this commented out to be fixed in another PR? If not, we should delete it.

File: cypress/support/hostsfunctions.ts [hunk 2]
```diff
@@ -322,3 +335,30 @@ export function checkFactsInHost(inventory: Inventory, hostInInventory?: boolean
...
+  //in case this should be done to detailes view the only thing needed is to add click on host name
+  navigateToBaseView(host_type, inventory.name);
+  cy.get(`[aria-label="Simple table"] tr`).its('length').should('be.gt', 1);
+  cy.get(missing).should('not.exist');
+}
+
+export function testMissingTab(
+  host_type: string,
+  inventory: Inventory,
+  //view: string,
```
Comment (fullsushidev) [inline cypress/support/hostsfunctions.ts:355]: same as before

**Review by fullsushidev — CHANGES_REQUESTED** [review]

File: cypress/support/hostsfunctions.ts (hunk 1 above)
Comment (RedOrNot) [reply cypress/support/hostsfunctions.ts]: done

File: cypress/support/hostsfunctions.ts (hunk 2 above)
Comment (RedOrNot) [reply cypress/support/hostsfunctions.ts]: done

File: cypress/e2e/awx/resources/inventoryHost/inventoryHostConstructed.cy.ts [hunk 3]
```diff
@@ -4,6 +4,7 @@ import { Inventory } from '../../../../../frontend/awx/interfaces/Inventory';
 import { InventoryGroup } from '../../../../../frontend/awx/interfaces/InventoryGroup';
 import { Organization } from '../../../../../frontend/awx/interfaces/Organization';
 import { runCommand } from './runCommandFunction';
+import { testMissingButton, testMissingTab } from '../../../../support/hostsfunctions';
```
Comment (fullsushidev) [inline cypress/e2e/awx/resources/inventoryHost/inventoryHostConstructed.cy.ts:7]:
```suggestion
import { checkHiddenButton, checkHiddenTab } from '../../../../support/hostsfunctions';
```

File: cypress/e2e/awx/resources/inventoryHost/inventoryHostConstructed.cy.ts [hunk 4]
```diff
@@ -117,9 +118,21 @@ describe('Inventory Host Tab Tests for contructed inventory', () => {
...
-  it.skip('confirm that edit host button is missing from the host tab list of an inventory', () => {});
-
-  it.skip('confirm that delete host button is missing from the host tab list of an inventory', () => {});
-
-  it.skip('confirm that facts tab is missing from a host inside an inventory', () => {});
+  it('test edit, delete buttons and facts tab are not present for constructed inventory host options', () => {
+    //'confirm that edit host button is missing from the host tab list of an inventory', () => {
+    //navigate to constructed inventory host list
+    //verify edit button is missing
+    testMissingButton('inventory_host', inventory, `[data-cy="edit-host"]`);
```
Comment (fullsushidev) [inline cypress/e2e/awx/resources/inventoryHost/inventoryHostConstructed.cy.ts:125]:
```suggestion
    checkHiddenButton('inventory_host', inventory, `[data-cy="edit-host"]`);
```

File: cypress/e2e/awx/resources/inventoryHost/inventoryHostConstructed.cy.ts [hunk 5]
```diff
@@ -117,9 +118,21 @@ describe('Inventory Host Tab Tests for contructed inventory', () => {
...
+
+    //'confirm that delete host button is missing from the host tab list of an inventory'
+    //navigate to constructed inventory host list
+    //verify action dropdown contain only delete host button is missing
+    const missingValue = `[data-cy="actions-column-cell"] [data-cy="actions-dropdown"]`;
+    testMissingButton('inventory_host', inventory, missingValue);
```
Comment (fullsushidev) [inline cypress/e2e/awx/resources/inventoryHost/inventoryHostConstructed.cy.ts:131]:
```suggestion
    checkHiddenButton('inventory_host', inventory, hiddenElement);
```

File: cypress/e2e/awx/resources/inventoryHost/inventoryHostConstructed.cy.ts [hunk 6]
```diff
@@ -117,9 +118,21 @@ describe('Inventory Host Tab Tests for contructed inventory', () => {
...
+
+    //'confirm that facts tab is missing from a host inside an inventory'
+    //navigate to constructed inventory host list, get to host
+    //verify facts tab is missing
+    testMissingTab('inventory_host', inventory, 'Facts');
```
Comment (fullsushidev) [inline cypress/e2e/awx/resources/inventoryHost/inventoryHostConstructed.cy.ts:136]:
```suggestion
    checkHiddenTab('inventory_host', inventory, 'Facts');
```

**Comment by vishsanghishetty** [comment]
@RedOrNot for the failures I see on your PR, I opened a PR for fixing the test that is failing, try to rebase once it merges
https://github.com/ansible/ansible-ui/pull/2577

File: cypress/e2e/awx/resources/inventoryHost/inventoryHostSmart.cy.ts [hunk 7]
```diff
@@ -2,7 +2,7 @@
 import { Inventory } from '../../../../../frontend/awx/interfaces/Inventory';
 import { Organization } from '../../../../../frontend/awx/interfaces/Organization';
 import { AwxUser } from '../../../../../frontend/awx/interfaces/User';
-//import { createAndEditAndDeleteHost} from '../../../support/hostsfunctions';
+import { testMissingButton, testMissingTab } from '../../../../support/hostsfunctions';
```
Comment (fullsushidev) [inline cypress/e2e/awx/resources/inventoryHost/inventoryHostSmart.cy.ts:5]:
```suggestion
import { checkHiddenButton, checkHiddenTab } from '../../../../support/hostsfunctions';
```

File: cypress/e2e/awx/resources/inventoryHost/inventoryHostSmart.cy.ts [hunk 8]
```diff
@@ -59,9 +59,21 @@ describe('Inventory Host Tab Tests for smart inventory', () => {
     //5) Cancel the job and assert that it has been canceled
   });
 
-  it.skip('confirm that edit host button is missing from the host tab list of an inventory', () => {});
+  it('test edit, delete and facts are not part of constracted inventory host options', () => {
+    //'confirm that edit host button is missing from the host tab list of an inventory'
+    //navigate to smart inventory host list
+    //verify edit button is missing
+    testMissingButton('inventory_host', inventory, `[data-cy="edit-host"]`);
```
Comment (fullsushidev) [inline cypress/e2e/awx/resources/inventoryHost/inventoryHostSmart.cy.ts:66]:
```suggestion
    checkHiddenButton('inventory_host', inventory, `[data-cy="edit-host"]`);
```

File: cypress/e2e/awx/resources/inventoryHost/inventoryHostSmart.cy.ts [hunk 9]
```diff
@@ -59,9 +59,21 @@ describe('Inventory Host Tab Tests for smart inventory', () => {
...
 
-  it.skip('confirm that delete host button is missing from the host tab list of an inventory', () => {});
+    //'confirm that delete host button is missing from the host tab list of an inventory', () => {
+    //navigate to smart inventory host list
+    //verify action dropdown contain only delete host button is missing
+    const missingValue = `[data-cy="actions-column-cell"] [data-cy="actions-dropdown"]`;
+    testMissingButton('inventory_host', inventory, missingValue);
```
Comment (fullsushidev) [inline cypress/e2e/awx/resources/inventoryHost/inventoryHostSmart.cy.ts:72]:
```suggestion
    checkHiddenButton('inventory_host', inventory, hiddenElement);
```

File: cypress/e2e/awx/resources/inventoryHost/inventoryHostSmart.cy.ts [hunk 10]
```diff
@@ -59,9 +59,21 @@ describe('Inventory Host Tab Tests for smart inventory', () => {
...
 
-  it.skip('confirm that facts tab is missing from a host inside an inventory', () => {});
+    //'confirm that facts tab is missing from a host inside an inventory', () => {
+    //navigate to constructed inventory host list, get to host
+    //verify facts tab is missing
+    testMissingTab('inventory_host', inventory, 'Facts');
```
Comment (fullsushidev) [inline cypress/e2e/awx/resources/inventoryHost/inventoryHostSmart.cy.ts:77]:
```suggestion
    checkHiddenTab('inventory_host', inventory, 'Facts');
```

**Comment by vishsanghishetty** [comment]
one minor suggestion, we should keep the branch name always relevant to the feature you are writing tests for and also in lowercase. thank you for your contribution!

File: cypress/support/hostsfunctions.ts [hunk 11]
```diff
@@ -322,3 +335,20 @@ export function checkFactsInHost(inventory: Inventory, hostInInventory?: boolean
   cy.containsBy('a', 'Facts').click();
   cy.get('code').should('contain', 'ansible_dns');
 }
+
+export function testMissingButton(host_type: string, inventory: Inventory, missing: string) {
```
Comment (fullsushidev) [inline cypress/support/hostsfunctions.ts:339]:
```suggestion
export function checkHiddenButton(host_type: string, inventory: Inventory, missing: string) {
```

File: cypress/support/hostsfunctions.ts [hunk 12]
```diff
@@ -322,3 +335,20 @@ export function checkFactsInHost(inventory: Inventory, hostInInventory?: boolean
...
+  //navigate to list view and check host existing
+  //in case this should be done to detailes view the only thing needed is to add click on host name
+  navigateToBaseView(host_type, inventory.name);
+  cy.get(`[aria-label="Simple table"] tr`).its('length').should('be.gt', 1);
+  cy.get(missing).should('not.exist');
+}
+
+export function testMissingTab(host_type: string, inventory: Inventory, missing: string) {
```
Comment (fullsushidev) [inline cypress/support/hostsfunctions.ts:347]:
```suggestion
export function checkHiddenTab(host_type: string, inventory: Inventory, missing: string) {
```

File: cypress/e2e/awx/resources/inventoryHost/inventoryHostSmart.cy.ts [hunk 13]
```diff
@@ -59,9 +59,21 @@ describe('Inventory Host Tab Tests for smart inventory', () => {
...
+    //'confirm that edit host button is missing from the host tab list of an inventory'
+    //navigate to smart inventory host list
+    //verify edit button is missing
+    testMissingButton('inventory_host', inventory, `[data-cy="edit-host"]`);
 
-  it.skip('confirm that delete host button is missing from the host tab list of an inventory', () => {});
+    //'confirm that delete host button is missing from the host tab list of an inventory', () => {
+    //navigate to smart inventory host list
+    //verify action dropdown contain only delete host button is missing
+    const missingValue = `[data-cy="actions-column-cell"] [data-cy="actions-dropdown"]`;
```
Comment (fullsushidev) [inline cypress/e2e/awx/resources/inventoryHost/inventoryHostSmart.cy.ts:71]:
```suggestion
    const hiddenElement = `[data-cy="actions-column-cell"] [data-cy="actions-dropdown"]`;
```

File: cypress/e2e/awx/resources/inventoryHost/inventoryHostConstructed.cy.ts [hunk 14]
```diff
@@ -117,9 +118,21 @@ describe('Inventory Host Tab Tests for contructed inventory', () => {
...
+  it('test edit, delete buttons and facts tab are not present for constructed inventory host options', () => {
+    //'confirm that edit host button is missing from the host tab list of an inventory', () => {
+    //navigate to constructed inventory host list
+    //verify edit button is missing
+    testMissingButton('inventory_host', inventory, `[data-cy="edit-host"]`);
+
+    //'confirm that delete host button is missing from the host tab list of an inventory'
+    //navigate to constructed inventory host list
+    //verify action dropdown contain only delete host button is missing
+    const missingValue = `[data-cy="actions-column-cell"] [data-cy="actions-dropdown"]`;
```
Comment (fullsushidev) [inline cypress/e2e/awx/resources/inventoryHost/inventoryHostConstructed.cy.ts:130]:
```suggestion
    const hiddenElement = `[data-cy="actions-column-cell"] [data-cy="actions-dropdown"]`;
```

**Review by fullsushidev — APPROVED** [review]
I believe we could try to update some names to make it more clear. :)

There are some choice of functions I believe it needs to be revised (i.e. how we use `cy.contains`), but this can be updated in the next PR.

**Comment by fullsushidev** [comment]
Sorry for the commit flood. GitHub UI wasn't very nice 😳
I've rebased the branch and will wait the green pipeline to merge it.
